sql/expression/boolean.go

```diff
@@ -37,10 +37,37 @@ func (e *Not) Type() sql.Type {
 
 // Eval implements the Expression interface.
 func (e *Not) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
-	v, err := e.Child.Eval(ctx, row)
+	var v interface{}
+	var err error
+
+	// NOT-BETWEEN() is not equal to NOT(BETWEEN()) as cases with NULL value for min and/or max are
+	// handled differently than only negating the BETWEEN() result.
+	switch e.Child.(type) {
+	case *Between:
+		e.Child.(*Between).Negated = true
```

Comment:
You should get rid of this block of code. 

If you think this is the best way to implement NOT BETWEEN then in parse.go you should just return a BETWEEN object and remove the use of NOT